jsettlers.common/src/jsettlers/common/map/partition/IPartitionData.java

```diff
@@ -41,4 +41,13 @@
 	 * @return Returns the {@link EMaterialType} with the given priority.
 	 */
 	EMaterialType getMaterialTypeForPrio(int priorityIdx);
+
+	/**
+	 * Returns the number of materials of the given type, available in the respective partition.
+	 * 
+	 * @param materialType
+	 *            Type of material.
+	 * @return Available amount of that material.
+	 */
+	int getAmountOf(EMaterialType materialType);

```

Comment:
@adamjryan: You can use this method to get the number of materials available in the current partition. To get an instance of this interface, have a look at `IGraphicsGrid.getPartitionData(int x, int y):IPartitionData`.

At the moment `getAmountOf()` always returns 0. But I will implement it soon. In the meantime, you can start wiring it into the UI.

Furthermore, the method `IPartitonData.getDistributionSettings(EMaterialType materialType)` provides the material distribution for the buildings (i.e. what percentage of a material is going to each type of consumer building). This method should provide correct values already.